File: java/src/main/java/com/voteU/election/java/model/PollingStation.java
package com.voteU.election.java.model;

import lombok.Getter;
import lombok.Setter;

import java.util.HashMap;
import java.util.Map;

@Setter
@Getter
public class PollingStation {

    private String id;
    private String name;
    private String zipCode;
    private int votes;
    private String electionId;
    private String authorityId;
    private Map<Integer, Party> parties;

    public PollingStation(String id, String name, String zipCode) {
        this.id = id;
        this.name = name;
        this.zipCode = zipCode;
        this.votes = 0;
        this.parties = new HashMap<>();
        this.authorityId = "";
    }


    @Override
    public String toString() {
        return "PollingStation{" +
                "id='" + id + '\'' +
                ", name='" + name + '\'' +
                ", zipCode='" + zipCode + '\'' +
                ", parties=" + parties +
                '}';
    }
}

File: voteu-project/src/interface/PollingStation.ts
import type { Party } from '@/interface/Party.ts'

export interface PollingStation {
  id: string; // Unique ID of the authority (e.g., "0809")
  name: string; // Name of the authority
  zipCode: string;
  parties: {[partyId: number]: Party}// Map of parties with partyId as key
  authorityId: string
}
